src/app/client/src/app/modules/resource/components/collection-player/collection-player.component.ts

```diff
@@ -142,10 +142,23 @@ export class CollectionPlayerComponent implements OnInit, OnDestroy {
         id: content.metadata.identifier,
         type: content.metadata.contentType || content.metadata.resourceType || 'content',
         ver: content.metadata.pkgVersion || '1',
-        rollup: {l1: this.collectionId}
+        rollup: { l1: this.collectionId }
         // rollup: this.collectionInteractObject
       };
       this.triggerContentImpression = true;
+      // if mimeType is text/x-url extcontentpreview plugin will be invoked
+      if (content.metadata.mimeType === 'text/x-url') {
```

Comment:
why is this logic in initPlayer method? should it not be in playContent method?

Reply:
Collection player is down now, once it is up, i will test and move

Reply:
player config data is available inside the initPlayer function